d2graph/d2graph.go

```diff
@@ -968,6 +968,18 @@ func (obj *Object) GetDefaultSize(mtexts []*d2target.MText, ruler *textmeasure.R
 	return &dims, nil
 }
 
+func (obj *Object) IsInsideNearContainer() bool {
```

Comment:
have this return the near container object and call it `OuterNearContainer()` or `NearAncestor()`